src/Orleans/Runtime/OutsideRuntimeClient.cs

```diff
@@ -134,6 +134,9 @@ public OutsideRuntimeClient(ClientConfiguration cfg, GrainFactory grainFactory,
             logger = LogManager.GetLogger("OutsideRuntimeClient", LoggerType.Runtime);
             appLogger = LogManager.GetLogger("Application", LoggerType.Application);
 
+            BufferPool.InitGlobalBufferPool(config);
+            this.clientId = GrainId.NewClientId(cfg.ClusterId);
```

Comment:
>this.clientId = GrainId.NewClientId(cfg.ClusterId); [](start = 12, length = 51)

If you follow my suggestion to obtain ClusterId from a gateway silo from within `OutsideRuntimeClient.StartInternal()`, ClusterId won't be available here yet. Though is should be possible to send that one request with a temporary client ID without ClusterId in it yet, and then update it for further messaging.